### PR TITLE
feat: changed dropdown to checkbox in general-pref

### PR DESCRIPTION
**Pull request by prahaladhchandrahasan**
Continuing to make the UI better I have changed it from dropdown menu to a toggle button. 
Pattern fly Bootstrap switch: https://www.patternfly.org/v3/pattern-library/widgets/#bootstrap-switch is the ultimate goal.

This is an enhancement of the PR: https://github.com/hawtio/hawtio-core/pull/48

### CODE REVIEW

File: src/app/preferences/general-preferences/general-preferences.controller.ts [hunk 1]
```diff
@@ -1,18 +1,12 @@
-/// <reference path="general-preferences.service.ts"/>
-
 namespace Core {
-  export function GeneralPreferencesController($scope, generalPreferencesService: GeneralPreferencesService) {
+  export function GeneralPreferencesController($scope, $window: ng.IWindowService) {
```
Comment (tadayosi) [inline src/app/preferences/general-preferences/general-preferences.controller.ts:2]: Why did you delete the GeneralPreferencesService?   Understand the responsibilities of MVC model, where Model = service, View = html, and Controller = controller here. In MVC, only models (= services) should do any meaningful stuffs, such as persisting to local storage.

File: src/app/preferences/general-preferences/general-preferences.html [hunk 2]
```diff
@@ -2,12 +2,7 @@
   <form class="form-horizontal general-preferences-form">
     <div class="form-group">
       <label class="col-md-2 control-label" for="Enabling toggle">Hide vertical nav by default</label>
-      <div class="col-md-6">
-        <select id="vertical-nav" class="form-control" ng-model="defaultVerticalNavState"
-          ng-options="defaultVerticalNavState for defaultVerticalNavState in availableVerticalNavs.availableOptions"
-          ng-change="onToggleChange(defaultVerticalNavState)">
-        </select>
-      </div>
+      <input type="checkbox" ng-change="onToggleChange(defaultVerticalNavState)" ng-model="defaultVerticalNavState">
```
Comment (tadayosi) [inline src/app/preferences/general-preferences/general-preferences.html:5]: This doesn't seem to implement the bootstrap switch precisely enough.
https://www.patternfly.org/v3/pattern-library/widgets/#bootstrap-switch
And you'd need to invoke this somewhere in the controller.
```
jQuery(".bootstrap-switch").bootstrapSwitch();
```

Also see how checkbox can be implemented with AngularJS PF:
https://patternfly.github.io/angular-patternfly/#!/api/patternfly.validation:pfValidation

**Review by tadayosi — CHANGES_REQUESTED** [review]

**Comment by tadayosi** [comment]
And I think you can directly jump to bootstrap switch implementation. It shouldn't be so long leap.

File: src/app/preferences/general-preferences/general-preferences.controller.ts (hunk 1 above)
Comment (prahaladhchandrahasan) [reply src/app/preferences/general-preferences/general-preferences.controller.ts]: done

File: src/app/preferences/general-preferences/general-preferences.service.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ namespace Core {
       'ngInject';
     }
     getDefaultVerticalNavState(): string {
-      if (this.$window.localStorage.getItem('defaultVerticalNavState') !== null) {
+      if (this.$window.localStorage.getItem('defaultVerticalNavState') != undefined) {
```
Comment (tadayosi) [inline src/app/preferences/general-preferences/general-preferences.service.ts:8]: You can use `_.isNil(...)` here.
https://lodash.com/docs/4.17.15#isNull

This is also a good resource when you are considering to use lodash:
https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore

**Review by tadayosi — CHANGES_REQUESTED** [review]
Only a small change request but otherwise it looks good. GJ!

File: src/app/page/page.component.ts [hunk 4]
```diff
@@ -26,7 +26,7 @@ namespace Page {
       });
       this.$rootScope.$on(CLOSE_MAIN_NAV_EVENT, () => this.isNavOpen = false);
       const defaultVerticalNavState = this.$window.localStorage.getItem('defaultVerticalNavState');
-      if (defaultVerticalNavState === "show") {
+      if (defaultVerticalNavState == "show" || defaultVerticalNavState == null) {
```
Comment (tadayosi) [inline src/app/page/page.component.ts:29]: Always prefer to `===` instead of `==`.
https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons

Also this is better:
```
const defaultVerticalNavState = this.$window.localStorage.getItem('defaultVerticalNavState') || 'show';
if (defaultVerticalNavState === 'show') {
    ...
```

File: src/app/preferences/general-preferences/general-preferences.controller.ts [hunk 5]
```diff
@@ -1,20 +1,9 @@
...
-      availableOptions: [
-        "show", "hide"
-      ]
-    };
     $scope.defaultVerticalNavState = generalPreferencesService.getDefaultVerticalNavState();
-    $scope.onToggleChange = defaultVerticalNavState => {
-      if (defaultVerticalNavState !== null && defaultVerticalNavState !== undefined) {
-        generalPreferencesService.setDefaultVerticalNavState(defaultVerticalNavState);
-      }
+    $scope.onToggle= function() {
```
Comment (tadayosi) [inline src/app/preferences/general-preferences/general-preferences.controller.ts:5]: format the code.

File: src/app/preferences/general-preferences/general-preferences.service.ts [hunk 6]
```diff
@@ -5,9 +5,10 @@ namespace Core {
       'ngInject';
     }
     getDefaultVerticalNavState(): string {
-      if (this.$window.localStorage.getItem('defaultVerticalNavState') !== null) {
+      if (this.$window.localStorage.getItem('defaultVerticalNavState') != null) {
```
Comment (tadayosi) [inline src/app/preferences/general-preferences/general-preferences.service.ts:8]: Use `!==` here as well.

File: src/app/preferences/general-preferences/general-preferences.service.ts [hunk 7]
```diff
@@ -5,9 +5,10 @@ namespace Core {
...
         return this.$window.localStorage.getItem('defaultVerticalNavState');
       } else {
+        this.setDefaultVerticalNavState(GeneralPreferencesService.DEFAULT_VERTICAL_NAV_STATE);
```
Comment (tadayosi) [inline src/app/preferences/general-preferences/general-preferences.service.ts:11]: You don't need to persist the default value.

**Review by tadayosi — CHANGES_REQUESTED** [review]
Almost looks good. Just some nitpicky changes.

File: src/app/page/page.component.ts [hunk 8]
```diff
@@ -26,7 +26,7 @@ namespace Page {
       });
       this.$rootScope.$on(CLOSE_MAIN_NAV_EVENT, () => this.isNavOpen = false);
       const defaultVerticalNavState = this.$window.localStorage.getItem('defaultVerticalNavState');
-      if (defaultVerticalNavState === "show") {
+      if (defaultVerticalNavState === "show" || _.isNull(defaultVerticalNavState)) {
```
Comment (tadayosi) [inline src/app/page/page.component.ts:29]: no it is not what I suggest. Pls look at the code I presented:
```
const defaultVerticalNavState = this.$window.localStorage.getItem('defaultVerticalNavState') || 'show';
if (defaultVerticalNavState === 'show') {
    ...
```

**Review by tadayosi — CHANGES_REQUESTED** [review]
Only one comment.

**Review by tadayosi — APPROVED** [review]
GJ!
